airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -419,6 +430,31 @@ def create_declarative_stream(self, model: DeclarativeStreamModel, config: Confi
             parameters={},
         )
 
+    def _merge_stream_slicers(self, model: DeclarativeStreamModel, config: Config) -> Optional[StreamSlicer]:
+        incremental_sync = (
+            self._create_component_from_model(model=model.incremental_sync, config=config) if model.incremental_sync else None
+        )
+
+        stream_slicer = None
+        if model.retriever.stream_slicer:
+            stream_slicer_model = model.retriever.stream_slicer
+            stream_slicer = (
+                CartesianProductStreamSlicer(
+                    [self._create_component_from_model(model=slicer, config=config) for slicer in stream_slicer_model], parameters={}
+                )
+                if type(stream_slicer_model) == list
+                else self._create_component_from_model(model=stream_slicer_model, config=config)
+            )
+
+        if incremental_sync and stream_slicer:
+            return CartesianProductStreamSlicer(stream_slicers=[incremental_sync, stream_slicer], parameters=model.parameters)
+        elif incremental_sync:
+            return incremental_sync
+        elif stream_slicer:
+            return stream_slicer
+        else:
+            return None
```

Comment:
Is there a reason why we don't return `SingleSlice(parameters={})` here to avoid managing `Optional` here?

Reply:
Good question, I went back and forth on this one too. My thinking was that by putting it onto the retriever, it made it more clear that `SingleSlice()` was the default value of the `SimpleRetriever.stream_slicer` field. But that mean propagating the optional. No strong opinions either way though